amqprs/src/api/tls.rs

```diff
@@ -85,44 +89,63 @@ impl TlsAdaptor {
         let mut root_store = RootCertStore::empty();
         if let Some(root_ca_cert) = root_ca_cert {
             let mut pem = BufReader::new(File::open(root_ca_cert)?);
+
             let certs = rustls_pemfile::certs(&mut pem)?;
-            let trust_anchors = certs.iter().map(|cert| {
-                let ta = webpki::TrustAnchor::try_from_cert_der(&cert[..]).unwrap();
-                OwnedTrustAnchor::from_subject_spki_name_constraints(
-                    ta.subject,
-                    ta.spki,
-                    ta.name_constraints,
-                )
+
+            let trust_anchors = certs.into_iter().map(|cert| {
+                //                let c = webpki::Cert::from(cert);
+                let der = rustls_pki_types::CertificateDer::from(cert);
+                let anchor = webpki::anchor_from_trusted_cert(&der).unwrap().to_owned();
+
+                rustls_pki_types::TrustAnchor {
+                    subject: anchor.subject.into(),
+                    subject_public_key_info: anchor.subject_public_key_info.into(),
+                    name_constraints: anchor.name_constraints.map(|f| f.into()),
+                }
             });
-            root_store.add_server_trust_anchors(trust_anchors);
+
+            // NOTE: The old rustls add_server_trust_anchors function did this
+            // https://github.com/rustls/rustls/blob/d1345fc39ad597e27e6355341d2b2b40c501625b/rustls/src/anchors.rs#L117-L118
+            root_store.roots.extend(trust_anchors);
         } else {
-            root_store.add_server_trust_anchors(webpki_roots::TLS_SERVER_ROOTS.0.iter().map(
-                |ta| {
-                    OwnedTrustAnchor::from_subject_spki_name_constraints(
-                        ta.subject,
-                        ta.spki,
-                        ta.name_constraints,
-                    )
-                },
-            ));
+            root_store
+                .roots
+                .extend(webpki_roots::TLS_SERVER_ROOTS.0.iter().map(|ta| {
+                    rustls_pki_types::TrustAnchor {
+                        subject: ta.subject.into(),
+                        subject_public_key_info: ta.spki.into(),
+                        name_constraints: ta.name_constraints.map(|f| f.into()),
+                    }
+                }));
         }
         Ok(root_store)
     }
 
-    fn build_client_certificates(client_cert: &Path) -> std::io::Result<Vec<Certificate>> {
-        let mut pem = BufReader::new(File::open(client_cert)?);
-        let certs = rustls_pemfile::certs(&mut pem)?;
-        let certs = certs.into_iter().map(Certificate);
+    fn build_client_certificates(
+        client_cert: PathBuf,
```

Comment:
same, no need to use PathBuf

Reply:
Fixed by https://github.com/gftea/amqprs/pull/141/commits/2d126b98c506d47300904c83e7c6abc15a478395